### PR TITLE
Adding Intelligence to verify-io-on-site on post deletion objects

### PR DESCRIPTION
**Pull request by hmaheswa**
We have issue if conf file has both create_object and delete_bucket_object set to true, and verify-io-on-site raises "HeadObject operation: Not Found" Exception because deleted object is being verified here.
This PR fixes the issue of verify-io for deleted user, bucket and object.
logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-MSHOU9/
logs of happy path: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-MKOB1R/

Signed-off-by: Hemanth Maheswarla <hmaheswa@hmaheswa.remote.csb>

### CODE REVIEW

**Comment by mkasturi18** [comment]
@hmaheswa could you update the pass logs

**Comment by mkasturi18** [comment]
@hmaheswa , as discussed could you please attach the complete log, as this change would impact the existing workflow as well

File: rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py [hunk 1]
```diff
@@ -61,12 +61,12 @@
 encryption_key = hashlib.md5(password).hexdigest()
 
 
-def test_exec(config, ssh_con):
+def test_exec(config):
```
Comment (ckulal) [inline rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:64]: Any reason why ssh_con was removed

I think it was added as a part of PR: Adds support script execution from non-RGW node

Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:64]: yeah, not removed intentionally, will update the branch.

**Review by anrao19 — DISMISSED** [review]

**Review by ckulal — APPROVED** [review]

File: rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py [hunk 2]
```diff
@@ -564,8 +564,7 @@ def test_exec(config, ssh_con):
                 raise TestExecError("Failed to enable user")
         if config.delete_user:
             user_id = each_user["user_id"]
-            cmd = f"radosgw-admin user rm --uid='{user_id}'"
-            out = utils.exec_shell_cmd(cmd)
+            out = reusable.remove_user(each_user)
```
Comment (udaysk23) [inline rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:567]: Could you please check scripts available in our repos and change it with `reusable.remove_user(each_user)` where we are deleting user using cli

File: rgw/v2/lib/s3/write_io_info.py [hunk 3]
```diff
@@ -242,6 +282,71 @@ def add_keys_info(self, access_key, bucket_name, key_info):
...
+        ):
+            if k["name"] == key_name:
+                key_indx = i
+                break
+        yaml_data["users"][access_key_indx]["bucket"][bucket_indx]["keys"][key_indx][
+            "deleted"
+        ] = True
+        self.file_op.add_data(yaml_data)
+
+    def set_keys_deleted_in_bucket_with_prefix(
```
Comment (udaysk23) [inline rgw/v2/lib/s3/write_io_info.py:315]: Can we have one run with LC script and one with delete object script

Comment (hmaheswa) [reply rgw/v2/lib/s3/write_io_info.py:315]: uday, I have raised seperate PR for lc verify io. This PR is only for delete object, attached the logs of it.

File: rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py (hunk 2 above)
Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py:567]: done uday

**Review by udaysk23 — APPROVED** [review]

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**